### ISSUE TITLE
allow users to change username?

### ISSUE THREAD

**Issue by ChillarAnand** [pythonindia/junction#264]
if user signups using 3rd party account, username is generating automatically based on name.

**Comment by kracekumar** [comment]
Yes, this was already asked by one of the user. We should have this.

+1

On Sat, May 30, 2015 at 12:58 PM, Anand Reddy Pandikunta <
notifications@github.com> wrote:

> if user signups using 3rd party account, username is generating
> automatically based on name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pythonindia/junction/issues/264>.
>



-- 
Regards
Kracekumar Ramaraju
http://kracekumar.com
+91 85530 29521

**Comment by NabeelValapra** [comment]
how about setting ```SOCIALACCOUNT_AUTO_SIGNUP = False ``` ?

**Comment by NabeelValapra** [comment]
How about setting ```SOCIALACCOUNT_AUTO_SIGNUP = False``` ?

**Comment by ChillarAnand** [comment]
by setting that all auth will display a default form with email & username after verification from 3rd party. we have to override that form to display only username.

or we can have a profile page where user can change it anytime?

**Comment by theskumar** [comment]
@ChillarAnand how about having this option available at both the places.

**Comment by ChillarAnand** [comment]
@theskumar  :+1:

**Comment by kracekumar** [comment]
I would lean to have profile section where user can modify these auto generated stuff. Signup process should be really quick. 

So along with `Social Accounts` in the drop down we can have Edit Profile option. 

We need also bigger dashboard for users to see their talks and selected talks and tickets information etc in the future, we will have separate issue for that.

For now we can have Edit profile section alone.

**Comment by vnbang2003** [comment]
username are not meant to be changeable.  I propose to add field alias_name in profile page.
if alias_name value is entered we show it else show username.
This is much easier and consistent way.

@ChillarAnand @kracekumar : Let me know our thoughts ?

**Comment by kracekumar** [comment]
@vnbang2003 Makes sense to me. We can call it `Display Name`.

**Comment by vsham20** [comment]
I am working on this issue

**Comment by ChillarAnand** [comment]
@vsham20 Looks like we don't have custom user model yet. We are using default `User` provided by django. We have to create new model here.

**Comment by ChillarAnand** [comment]
Dashboard view https://github.com/pythonindia/junction/blob/master/junction/profiles/views.py

**Comment by vsham20** [comment]
@ChillarAnand hii...how can i setup this on my local machine. I see the documentation but i am getting some errors.

**Comment by ChillarAnand** [comment]
Please post the errors you are getting.

**Comment by vsham20** [comment]
@ChillarAnand here http://paste.openstack.org/show/483364/ i am getting this "InsecurePlatformWarning" and i cant access it via browser too

**Comment by ChillarAnand** [comment]
It is just a warning. You can ignore that if packages are installed successfully.

**Comment by vsham20** [comment]
@ChillarAnand cool ...and what about accessing it?

**Comment by ChillarAnand** [comment]
If  that step  is  completed,  all  packages are installed.
What do you want to access?

**Comment by ibrahim45** [comment]
@ChillarAnand I am Working on it!!

**Comment by ChillarAnand** [comment]
@iambibhas any updates?

**Comment by kracekumar** [comment]
@ChillarAnand it's not @iambibhas, it's @ibrahim45

**Comment by ibrahim45** [comment]
@ChillarAnand Is the display name have to be shown in navigation bar?

**Comment by ChillarAnand** [comment]
If user has set `display_name`, then show it in navigation bar.

**Comment by ibrahim45** [comment]
@ChillarAnand  Is it right to use django template tag for showing display_name? please suggest if you have any idea on this.

**Comment by ChillarAnand** [comment]
If it is stored in user model, `{{ user.display_name}}` should work?
